gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,238 @@
+    >>> project.number is None
+    True
+
+The ``project`` referenced above has no extra metadata associated with it,
+however you can still operate on it (ie, ``project.create()`` or
+``project.delete()``).
+
+If you want to retrieve a project and all of it's metadata, the best method
+to use is ``get_project()``, which will return ``None`` if the project
+doesn't exist::
+
+    >>> from gcloud import resource_manager
+    >>> client = resource_manager.Client()
+    >>> project = client.get_project('purple-spaceship-123')
+    >>> project.number is None
+    False
+    >>> project = client.get_project('doesnt-exist')
+    >>> project is None
+    True
+"""
+
+
+from gcloud.client import Client as BaseClient
+from gcloud.exceptions import NotFound
+from gcloud.iterator import Iterator
+from gcloud.resource_manager.connection import Connection
+from gcloud.resource_manager.project import Project
+
+
+class Client(BaseClient):
+    """Client to bundle configuration needed for API requests.
+
+    See
+    https://cloud.google.com/resource-manager/reference/rest/
+    for more information on this API.
+
+    Automatically get credentials::
+
+        >>> from gcloud.resource_manager import Client
+        >>> client = Client()
+
+    .. note::
+
+        Chances are you want to use either the constructor with no arguments,
+        or one of the factory methods (like
+        :func:`gcloud.resource_manager.client.Client.from_service_account_json`
+        or similar).
+
+        Even more likely is that you want to use the Cloud SDK to get
+        credentials for these API calls (that is, run ``gcloud auth login``).
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                       :class:`NoneType`
+    :param credentials: The OAuth2 Credentials to use for the connection
+                        owned by this client. If not passed (and if no ``http``
+                        object is passed), falls back to the default inferred
+                        from the environment.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+    """
+
+    _connection_class = Connection
+
+    def list_projects(self, filter_params=None, page_size=None):
+        """List the projects visible to this client.
+
+        Example::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> for project in client.list_projects():
+            ...     print project.project_id
+
+        List all projects with label ``'environment'`` set to ``'prod'``
+        (filtering by labels)::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> env_filter = {'labels.environment': 'prod'}
```

Comment:
Do we want to provide an interface for the possible filters? (Outside of this PR to keep scope small, but worth filing a bug if there is a well-defined list of filters.)